### PR TITLE
Use "iterable" type hint and is_iterable() check.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: src/Collection/Iterator/InsertIterator.php [hunk 1]
```diff
@@ -65,7 +65,7 @@ class InsertIterator extends Collection
      * @param array|\Traversable $values The source collection from which the values will
      * be inserted at the specified path.
      */
-    public function __construct($into, $path, $values)
+    public function __construct(iterable $into, $path, iterable $values)
```
Comment (markstory) [inline src/Collection/Iterator/InsertIterator.php]: Should we update the doc block types as well?

**Review by markstory — COMMENTED** [review]
NIce work.

File: src/Collection/Iterator/InsertIterator.php (hunk 1 above)
Comment (markstory) [reply src/Collection/Iterator/InsertIterator.php]: Lets find out!

Comment (ADmad) [reply src/Collection/Iterator/InsertIterator.php]: Done.

**Pull request merged**